src/Illuminate/Database/Eloquent/PendingHasThroughRelationship.php

```diff
@@ -10,6 +10,7 @@
 /**
  * @template TIntermediateModel of \Illuminate\Database\Eloquent\Model
  * @template TDeclaringModel of \Illuminate\Database\Eloquent\Model
+ * @template TLocalRelationship of \Illuminate\Database\Eloquent\Relations\HasOneOrMany<TIntermediateModel, TDeclaringModel>
```

Comment:
I'm not sure if you need all three templates---you might can do with just the relation template given that both the model types can be gleaned from the relation type

Reply:
Is there some canonical way to do this? Last time I dived into this I was told there to be no trivial way to do such a thing; see https://github.com/phpstan/phpstan/discussions/8814.